### PR TITLE
Resolve differences between my merge and Olivier's

### PR DESCRIPTION
**Pull request by jnm**
## Description

Two developers independently resolved conflicts in a complex merge and then compared their results.

### CODE REVIEW

File: kpi/tests/api/v2/test_api_assets.py [hunk 1]
```diff
@@ -601,17 +601,19 @@ def setUp(self):
         self.client.login(username='someuser', password='someuser')
         self.current_username = 'someuser'
         self.asset = Asset.objects.filter(owner__username='someuser').first()
-        self.list_url = reverse(
-            self._get_endpoint('asset-file-list'), args=[self.asset.uid]
-        )
+        self.list_url = reverse(self._get_endpoint('asset-file-list'), args=[self.asset.uid])
```
Comment (noliveleger) [inline kpi/tests/api/v2/test_api_assets.py:604]: 🤔  I'm wondering why `Black` did not suggest the format I used over yours. This line seems to go beyond the 80 characters limit.

File: kpi/tests/test_permissions.py [hunk 2]
```diff
@@ -5,17 +5,17 @@
...
-    PERM_VALIDATE_SUBMISSIONS,
-    PERM_DISCOVER_ASSET,
     PERM_DELETE_ASSET,
     PERM_DELETE_SUBMISSIONS,
+    PERM_DISCOVER_ASSET,
+    PERM_MANAGE_ASSET,
     PERM_PARTIAL_SUBMISSIONS,
+    PERM_VALIDATE_SUBMISSIONS,
+    PERM_VIEW_ASSET,
+    PERM_VIEW_SUBMISSIONS,
```
Comment (noliveleger) [inline kpi/tests/test_permissions.py:18]: ❤️

File: kpi/utils/object_permission_helper.py [hunk 3]
```diff
@@ -78,7 +64,7 @@ def get_user_permission_assignments(cls, affected_object, user,
 
         if not user or user.is_anonymous:
             filtered_user_ids = [affected_object.owner_id]
-        elif not cls.user_can_share(affected_object, user):
+        elif not affected_object.has_perm(user, PERM_MANAGE_ASSET):
```
Comment (noliveleger) [inline kpi/utils/object_permission_helper.py:67]: So obvious ;-)

**Review by noliveleger — APPROVED** [review]
Looks great!

File: kpi/utils/object_permission_helper.py (hunk 3 above)
Comment (jnm) [reply kpi/utils/object_permission_helper.py:67]: I almost left `user_can_share()`, but I decided to follow through on my `TODO` :smile:

File: kpi/tests/api/v2/test_api_assets.py (hunk 1 above)
Comment (jnm) [reply kpi/tests/api/v2/test_api_assets.py:604]: I was mistaken to include this change in this PR :frowning: I don't think I'll make a separate commit or PR to fix it, though. It'll just be the next person in here who fixes my sloppiness :smirk:

File: kpi/utils/object_permission_helper.py (hunk 3 above)
Comment (noliveleger) [reply kpi/utils/object_permission_helper.py:67]: Wise decision ;-)

File: kpi/tests/api/v2/test_api_assets.py (hunk 1 above)
Comment (noliveleger) [reply kpi/tests/api/v2/test_api_assets.py:604]: I thought exactly the same :-)

**Pull request merged**